src/Unleash/DefaultUnleash.cs

```diff
@@ -100,13 +100,15 @@ public bool IsEnabled(string toggleName, UnleashContext context)
 
         public bool IsEnabled(string toggleName, UnleashContext context, bool defaultSetting)
         {
-            return CheckIsEnabled(toggleName, context, defaultSetting);
+            return CheckIsEnabled(toggleName, context, defaultSetting).Enabled;
         }
 
-        private bool CheckIsEnabled(string toggleName, UnleashContext context, bool defaultSetting)
+        private FeatureEvaluationResult CheckIsEnabled(string toggleName, UnleashContext context, bool defaultSetting, Variant defaultVariant = null)
```

Comment:
It's a matter of personal preferences but I'd add some chunking to reason about the algorithm steps:
```c#
private FeatureEvaluationResult CheckIsEnabled(string toggleName, UnleashContext context, bool defaultSetting, Variant defaultVariant = null)
{
    var featureToggle = GetToggle(toggleName);
    var enhancedContext = context.ApplyStaticFields(settings);
    ActivationStrategy strategy = DetermineStrategy(featureToggle, enhancedContext);
    bool enabled = DetermineEnabledState(featureToggle, defaultSetting, strategy);
    Variant variant = DetermineVariant(enabled, strategy, featureToggle, context, defaultVariant);

    RegisterCount(toggleName, enabled);

    if (featureToggle?.ImpressionData ?? false) 
        EmitImpressionEvent("isEnabled", enhancedContext, enabled, featureToggle.Name);

    return new FeatureEvaluationResult { Enabled = enabled, Variant = variant };
}

private ActivationStrategy DetermineStrategy(FeatureToggle featureToggle, UnleashContext enhancedContext)
{
    if (featureToggle == null || !featureToggle.Enabled || featureToggle.Strategies.Count == 0)
        return null;

    return featureToggle.Strategies.FirstOrDefault(s =>
        GetStrategyOrUnknown(s.Name).IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints)));
}

private bool DetermineEnabledState(FeatureToggle featureToggle, bool defaultSetting, ActivationStrategy strategy)
{
    if (featureToggle == null) return defaultSetting;
    if (!featureToggle.Enabled) return false;
    if (featureToggle.Strategies.Count == 0) return true;
    return strategy != null;
}

private Variant DetermineVariant(bool enabled, ActivationStrategy strategy, FeatureToggle featureToggle, UnleashContext context, Variant defaultVariant)
{
    if (!enabled) return defaultVariant;
    
    if (strategy != null)
    {
        string groupId = null;
        strategy.Parameters.TryGetValue("groupId", out groupId);
        return VariantUtils.SelectVariant(groupId ?? featureToggle.Name, context, strategy.Variants);
    }

    return defaultVariant ?? VariantUtils.SelectVariant(featureToggle, context, defaultVariant);
}
```

Reply:
Thank you, I've used this for inspiration but opted to keep Enabled and Strategy checks in the same method for long term consistency in enabled-checks